torchvision/prototype/datasets/utils/_dataset.py

```diff
@@ -196,7 +196,18 @@ def _verify_str_arg(
     ) -> str:
         return verify_str_arg(value, arg, valid_values, custom_msg=custom_msg)
 
-    def __init__(self, root: Union[str, pathlib.Path], *, skip_integrity_check: bool = False) -> None:
+    def __init__(
+        self, root: Union[str, pathlib.Path], *, skip_integrity_check: bool = False, dependencies: Collection[str] = ()
+    ) -> None:
+        for dependency in dependencies:
+            try:
+                importlib.import_module(dependency)
+            except ModuleNotFoundError as error:
+                raise ModuleNotFoundError(
+                    f"{type(self).__name__}() depends on the third-party package '{dependency}'. "
+                    f"Please install it, for example with `pip install {dependency}`."
```

Comment:
Are we sure all packages are available on PyPI? And are we sure these will be compatible with the user's setup? I'd rather not give explicit instrutions here

Reply:
> Are we sure all packages are available on PyPI?

I think so yes. I don't think we can depend on packages that are not hosted on PyPI for our public datasets.

> And are we sure these will be compatible with the user's setup? I'd rather not give explicit instrutions here

No, we can't be sure. Given that wheels on PyPI are self-contained, I'm guessing we should be fine. The intention here is to help new users to get started. If we can expect them to handle the installation themselves, I'm ok to remove the instructions.

Reply:
Ok let's leave it as is, we can always revisit